core/src/main/java/io/aiven/kafka/tieredstorage/manifest/serde/DataKeySerializer.java

```diff
@@ -19,24 +19,32 @@
 import javax.crypto.SecretKey;
 
 import java.io.IOException;
+import java.util.Base64;
+import java.util.Objects;
 import java.util.function.Function;
 
+import io.aiven.kafka.tieredstorage.security.DataKeyEncryptionResult;
+
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.databind.SerializerProvider;
 import com.fasterxml.jackson.databind.ser.std.StdSerializer;
 
 public class DataKeySerializer extends StdSerializer<SecretKey> {
-    private final Function<SecretKey, byte[]> keyEncryptor;
+    private final Function<SecretKey, DataKeyEncryptionResult> dataKeyEncryptor;
 
-    public DataKeySerializer(final Function<SecretKey, byte[]> keyEncryptor) {
+    public DataKeySerializer(final Function<SecretKey, DataKeyEncryptionResult> dataKeyEncryptor) {
         super(SecretKey.class);
-        this.keyEncryptor = keyEncryptor;
+        this.dataKeyEncryptor = Objects.requireNonNull(dataKeyEncryptor, "dataKeyEncryptor cannot be null");
     }
 
     @Override
     public void serialize(final SecretKey value,
                           final JsonGenerator gen,
                           final SerializerProvider provider) throws IOException {
-        gen.writeBinary(keyEncryptor.apply(value));
+        final DataKeyEncryptionResult encryptionResult = dataKeyEncryptor.apply(value);
+        gen.writeString(
+            encryptionResult.keyEncryptionKeyId
+            + ":"
+            + Base64.getEncoder().encodeToString(encryptionResult.encryptedDataKey));
```

Comment:
Following suggestion to move decoding to DataEncryptionKeyResult:
```suggestion
        gen.writeString(encryptionResult.encode());
```

Reply:
Yes, done this. Also added more constructor checks and tests.